client/src/util/dataframe/labelIndex.ts

```diff
@@ -383,48 +386,27 @@ class KeyIndex {
 
     this.index = index;
     this.rindex = rindex;
-    this.__id = __getMemoId();
-    this.__compile();
-  }
-
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
-  __compile() {
-    const { index, rindex } = this;
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    this.getOffset = function getOffset(k: any) {
-      return index.get(k);
-    };
-
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    this.getOffsets = function getOffsets(arr: any) {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      return arr.map((l: any) => this.getOffset(l));
-    };
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    this.getLabel = function getLabel(i: any) {
-      return Number.isInteger(i) ? rindex[i] : undefined;
+    this.getOffset = function getOffset(label: LabelType) {
+      const offset = index.get(label);
+      if (offset === undefined) return -1;
+      return offset;
     };
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-    this.getLabels = function getLabels(arr: any) {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      return arr.map((i: any) => this.getLabel(i));
+    this.getLabel = function getLabel(offset: number) {
+      return Number.isInteger(offset) ? rindex[offset] : undefined;
```

Comment:
same question here for `undefined` vs. `-1`

Reply:
same answer as above. See the abstract method definition.